### PR TITLE
fix(curriculum): Statistics Calculator make anonymous callback function optional

### LINKED ISSUE #54997: Statistics calculator - Step 5: should not require an anonymous callback
### Describe the Issue

I don't think this should require an anonymous callback when you can just pass `Number` as the callback. It should pass with both.

Expected:
```
const numbers = array.map((el) => Number(el));
```

Valid code not passing:

```
const numbers = array.map(Number);
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-advanced-array-methods-by-building-a-statistics-calculator/step-5

### Your code

```



```

### Expected behavior

Valid code should pass.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Supravisor**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54997

image

image

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md [hunk 1]
```diff
@@ -24,13 +24,13 @@ Add a callback function to your `.map()` method that converts each element to a
 Your `.map()` method should have a callback function which takes an `el` parameter.
 
 ```js
-assert.match(calculate.toString(), /array\.map\(\s*(\(\s*el\s*\)|el)\s*=>|array\.map\(\s*function\s*\(\s*el\s*\)\s*\{/)
+assert.match(calculate.toString(), /array\.map\(\s*(\(\s*el\s*\)|el)\s*=>|array\.map\(\s*function\s*\(\s*el\s*\)\s*\{|array\.map\(\s*Number\s*\)/)
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md:27]:
```suggestion
assert.match(calculate.toString(), /array\.map\(\s*(\(\s*\w+\s*\)|\w+)\s*=>|array\.map\(\s*function\s*\(\s*\w+\s*\)\s*\{|array\.map\(\s*Number\s*\)/)
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md [hunk 2]
```diff
@@ -24,13 +24,13 @@ Add a callback function to your `.map()` method that converts each element to a
...
 ```
 
 Your callback function should use the `Number` constructor to convert `el` to a number.
 
 ```js
-assert.match(calculate.toString(), /Number\(\s*el\s*\)/);
+assert.match(calculate.toString(), /Number\(\s*el\s*\)|Number/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md:33]:
```suggestion
assert.match(calculate.toString(), /Number\(\s*\w+\s*\)|Number/);
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ assert.notMatch(calculate.toString(), /new/);
 Your callback function should return the element converted to a number.
 
 ```js
-assert.match(calculate.toString(), /(array\.map\(\s*(\(\s*el\s*\)|el)\s*=>|array\.map\(\s*function\s*\(\s*el\s*\)\s*\{)\s*(return\s+)?Number\(\s*el\s*\)/);
+assert.match(calculate.toString(), /(array\.map\(\s*(\(\s*el\s*\)|el)\s*=>|array\.map\(\s*function\s*\(\s*el\s*\)\s*\{)\s*(return\s+)?Number\(\s*el\s*\)|array\.map\(\s*Number\s*\)/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/63507c4b63731437227b0134.md:45]:
```suggestion
assert.match(calculate.toString(), /(array\.map\(\s*(\(\s*\w+\s*\)|\w+)\s*=>|array\.map\(\s*function\s*\(\s*\w+\s*\)\s*\{)\s*(return\s+)?Number\(\s*\w+\s*\)|array\.map\(\s*Number\s*\)/);
```

**Review by lasjorg — CHANGES_REQUESTED** [review]
The hint text is wrong now. It should not require a `el` parameter, as that is only when using an anonymous callback function.

As such, we really should also not require a specific parameter name, but any valid identifier. I think we can just use `\w` and ignore that it will match on invalid identifier names, as that will cause a syntax error anyway.

**Review by lasjorg — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**